src/chocolatey/infrastructure.app/commands/ChocolateyInfoCommand.cs

```diff
@@ -38,7 +38,10 @@ public override void ConfigureArgumentParser(OptionSet optionSet, ChocolateyConf
                 .Add(
                     "s=|source=",
                     "Source - Source location for install. Can use special 'windowsfeatures', 'ruby', 'cygwin', or 'python' sources. Defaults to configured sources.",
-                    option => configuration.Sources = option.UnquoteSafe())
+                    option => {
+                        configuration.Sources = configuration.ExplicitSources = option.UnquoteSafe();
+                        configuration.ListCommand.ExplicitSource = true;

```

Comment:
Why are we storing a `Source` and an `ExplicitSource`, at the same time that we are setting a boolean property called `ExplicitSource`?

Also, what happens if the source that is passed is the named source of a configuration? Or one of the alternative sources?
Should that also be considered an explicit source?

Reply:
In the credential provider, the first thing it does it try to look up any explicitly set named sources, and then the remainder as URIs. I if it's an alternative source it'll be handled before getting to the credential provider and so will not matter if it's set as an explicit source.

Reply:
I am ok with the introduction of the `ExplicitSources` property on the top-level configuration object, but that does bring up the question of whether the `ExplicitSource` boolean is needed anymore on the ListCommand object.  Should we not replace the usage of `ExplicitSource` in the `ChocolateyListCommand` class, and replace if with having anything in the `ExplicitSources` property?  Perhaps mark `ExplicitSource` as obsolete, and then we can remove it in future breaking change?

Reply:
> Should we not replace the usage of `ExplicitSource` in the `ChocolateyListCommand` class, and replace if with having anything in the `ExplicitSources` property? Perhaps mark `ExplicitSource` as obsolete, and then we can remove it in future breaking change?

If I recall my conversation with Rain correctly, one thing they thought of for `ExplicitSource` on the `ChocolateyListCommand` class would be to do it as a calculated property and return true if the top-level `ExplicitSources` is set. But that change can't be introduced right now.

Reply:
That makes sense. Neither Extension nor GUI actually make use of the `ListCommand.ExplicitSource` setting. So if I'm understanding correctly, the idea is to: mark `ListCommand.ExplicitSource` Obsolete, and change where it's used in CLI code.

Reply:
Yes, that is what I am suggesting.

Reply:
I have removed the additions where we were setting this property in the new places, and marked the property obsolete.